src/HideLooseAssetService.cpp

```diff
@@ -19,12 +22,35 @@ namespace BsaPacker
 			return false;
 		}
 
+		// Only hide loose files that were not excluded when creating the archive (blacklisted), so might still be needed by the mod
+		QStringList blacklistExtensions = this->m_SettingsService->GetPluginSetting(SettingsService::SETTING_BLACKLISTED_FILES).toString().split(';');
+		for (auto& ext : blacklistExtensions) {
+			ext.prepend("*");
+		}
+
 		const QString& absModDir = modDirectory.absolutePath();
 		for (const QString& subDir : modDirectory.entryList(QDir::Dirs | QDir::NoDotAndDotDot)) {
+			// Hide subdirectories
 			const QString& absPath = absModDir + '/' + subDir;
-			QDir dir(absPath);
-			if (!dir.dirName().endsWith(".mohidden")) {
-				dir.rename(absPath, absPath + ".mohidden");
+			QDir originalDir(absPath);
+			if (originalDir.dirName().endsWith(s_HiddenExt) || originalDir.isEmpty()) {
+				continue;
+			}
+			if (!originalDir.rename(originalDir.absolutePath(), originalDir.absolutePath() + s_HiddenExt)) {
+				qWarning() << "Failed to hide " << originalDir.absolutePath();
+				continue;
+			}
+
+			// Restore files with blacklisted extension to their original directories
+			QDir hiddenDir(originalDir.absolutePath() + s_HiddenExt);
+			QDirIterator iterator(hiddenDir.absolutePath(), blacklistExtensions, QDir::Files, QDirIterator::Subdirectories);
+			while (iterator.hasNext()) {
+				QString hiddenFilePath = iterator.next();
+				QString originalFilePath = originalDir.absoluteFilePath(hiddenDir.relativeFilePath(hiddenFilePath));
+				originalDir.mkpath(originalFilePath.left(originalFilePath.lastIndexOf("/")));
```

Comment:
Ideally we would always use QFileInfo and QDir to do path operations, but the original code is guilty of this too.